Src/VimCore/SearchService.fs

```diff
@@ -199,6 +193,20 @@ type internal EditorSearchService
 
         result
 
+    /// Get the match, if any, at the specified position
+    member x.GetMatchAtPosition (findData: FindData) (position: int) =
+        let snapshot = findData.TextSnapshotToSearch
+        let checkOptions = findData.FindOptions &&& (~~~FindOptions.SearchReverse)
+        let checkFindData = FindData(findData.SearchString, snapshot, checkOptions, findData.TextStructureNavigator)
+        match x.FindNextRaw checkFindData position with
+        | Some matchSpan ->

```

Comment:
Consider using `when` here. I think it makes it a bit more concise. I've actually becoming a bigger fan of `when` after reviewing your PRs.

Reply:
Part of my goal for contributing to VsVim is to get more real-world experience programming F#. One of the side effects of writing a lot of code is gradually learning about language features and formulating opinions about them.

The nuance I now understand about `when` with `match is that it subverts "exhaustive matching". That is fine when you are trying to do a match for one thing with an additional condition. But it always has to be weighed against the security of exhaustive matching.

In this case, `Some`/`None` matching will never be expanded to include a third option and so it is a good candidate for `match`/`when`.